drivers/platform/maxim/max32650/maxim_spi.c

```diff
@@ -256,7 +257,22 @@ static int _max_spi_config(struct no_os_spi_desc *desc)
 	if (ret)
 		goto err_init;
 
-	ret = MXC_SPI_SetMode(MXC_SPI_GET_SPI(desc->device_id), desc->mode);
+	/* For Maxim Platforms SPI Mode 1 and 2 are reversed */
+	switch (desc->mode) {
+	case NO_OS_SPI_MODE_1:
+		mode = SPI_MODE_2;
+		break;
+	case NO_OS_SPI_MODE_2:
+		mode = SPI_MODE_1;
+		break;
+	case NO_OS_SPI_MODE_0:
+	case NO_OS_SPI_MODE_3:
+		mode = (mxc_spi_mode_t)desc->mode;
```

Comment:
this returns `-EINVAL` for SPI modes 0 and 3 (needs a break). Also, adding a `/* fallthrough */` comment will get rid of the warning.